### PR TITLE
jets: parse

### PR DESCRIPTION
**Pull request by matthew-levan**

### CODE REVIEW

File: rust/ares/src/jets/parse.rs [hunk 1]
```diff
@@ -261,7 +261,7 @@ pub fn jet_here(context: &mut Context, subject: Noun) -> Result {
     let p_vex = vex.head();
     let q_vex = vex.tail();
 
-    if unsafe { q_vex.raw_equals(D(0)) } {
+    if q_vex.is_atom() {
```
Comment (ashelkovnykov) [inline rust/ares/src/jets/parse.rs:264]: This is a jet mismatch:
```
?~  q.vex
  vex
...
```

Comment (matthew-levan) [reply rust/ares/src/jets/parse.rs:264]: We're preserving the logic from the Vere jet:
```
    if ( c3n == u3du(q_vex) ) {
      return vex;
    }
```

I forgot, what's the policy here? Preserve the mismatch?

Comment (matthew-levan) [reply rust/ares/src/jets/parse.rs]: Ok, fixed mismatch.

**Review by ashelkovnykov — APPROVED** [review]

**Comment by ashelkovnykov** [comment]
For posterity, test benchmarking on my machine as of Dec 8:
```
Vere serf w/ full pill:                 3m  6s
Vere serf w/ slim pill:                 1m  5s
Ares serf w/ slim pill:                12m 25s
Ares serf w/ slim pill & parser jets:  12m  1s
```

**Pull request merged**